cpp/include/gar/utils/expression.h

```diff
@@ -0,0 +1,293 @@
+/** Copyright 2023 Alibaba Group Holding Limited.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <memory>
+#include <string>
+
+#include "arrow/compute/api.h"
+
+#include "gar/graph_info.h"
+
+#ifndef GAR_UTILS_EXPRESSION_H_
+#define GAR_UTILS_EXPRESSION_H_
+
+namespace GAR_NAMESPACE_INTERNAL {
+
+using ArrowExpression = arrow::compute::Expression;
+
+/**
+ * This class wraps an arrow::compute::Expression and provides methods for
+ * reading arrow::compute::Expression objects
+ */
+class Expression {
+ public:
+  Expression() = default;
+  Expression(const Expression& other) = default;
+  virtual ~Expression() = default;
+
+  /**
+   * @brief Evaluate Expression as arrow::compute::Expression e.g. new
+   * ExpressionEqual(new ExpressionProperty("a"), new
+   * ExpressionLiteral(1)) will be parsed as
+   * arrow::compute::equal(arrow::compute::field_ref("a"),
+   * arrow::compute::literal(1))
+   *
+   * @return The arrow::compute::Expression instance
+   */
+  virtual ArrowExpression Evaluate() = 0;
+};
+
+class ExpressionProperty : public Expression {
```

Comment:
It would be better to add some brief descriptions for the classes defined in this file.

Besides, if you think it is necessary, you can add new classes or methods into API references by updating [this file](https://github.com/alibaba/GraphAr/blob/main/docs/reference/api-reference-cpp.rst)

Reply:
Greate tips! let me take a look.